cmd/askd/handlers/form.go

```diff
@@ -102,3 +107,216 @@ func (formHandle) Delete(c *web.Context) error {
+// 200 Success, 400 Bad Request, 404 Not Found, 500 Internal
+func (formHandle) Aggregate(c *web.Context) error {
+	id := c.Params["form_id"]
+
+	aggregations, err := form.AggregateFormSubmissions(c.SessionID, c.Ctx["DB"].(*db.DB), id)
+	if err != nil {
+		return err
+	}
+
+	ak := AggregationKeys{
+		Aggregations: aggregations,
+	}
+
+	c.Respond(ak, http.StatusOK)
+
+	return nil
+}
+
+// AggregateGroup performs all aggregations across a form's submissions and returns a single group.
+// 200 Success, 400 Bad Request, 404 Not Found, 500 Internal
+func (formHandle) AggregateGroup(c *web.Context) error {
+	id := c.Params["form_id"]
+
+	aggregations, err := form.AggregateFormSubmissions(c.SessionID, c.Ctx["DB"].(*db.DB), id)
+	if err != nil {
+		return err
+	}
+
+	aggregation, ok := aggregations[c.Params["group_id"]]
+	if !ok {
+		c.Respond(nil, http.StatusNotFound)
+	}
+
+	c.Respond(aggregation, http.StatusOK)
+
+	return nil
+}
+
+//==============================================================================
+
+// Form Digests: this section contains transport/bluprints for sending digests of form/question
+// information to clients that do not need/won't understand full, vebose feeds.
+
+// FormQuestionOptionDigest is the blueprint for a single multiple choice option.
+type FormQuestionOptionDigest struct {
+	ID    string `json:"id" bson:"id"`
+	Value string `json:"value" bson:"value"`
+}
+
+// FormQuestionDigest is the bluprint for a question in a form group.
+type FormQuestionDigest struct {
+	ID      string                     `json:"id" bson:"id"`
+	Type    string                     `json:"type" bson:"type"`
+	Title   string                     `json:"title" bson:"title"`
+	GroupBy bool                       `json:"group_by" bson:"group_by"`
+	Options []FormQuestionOptionDigest `json:"options,omitempty" bson:"options,omitempty"`
+	Order   int                        `json:"order" bson:"order"`
+}
+
+// FormDigest is the blueprint for how we send form digests to clients.
+type FormDigest struct {
+	Questions map[string]FormQuestionDigest `json:"questions" bson:"questions"`
+}
+
+// Digest returns a form digest.
+// 200 Success, 400 Bad Request, 404 Not Found, 500 Internal
+func (formHandle) Digest(c *web.Context) error {
+	id := c.Params["form_id"]
+
+	// Load the form requested.
+	f, err := form.Retrieve(c.SessionID, c.Ctx["DB"].(*db.DB), id)
+	if err != nil {
+		return err
+	}
+
+	// Create a container for the question digests.
+	questions := make(map[string]FormQuestionDigest)
+
+	// Order is a counter to set the order questions in the form.
+	order := 1
+
+	// Loop through to form's steps/widgets to find the questions.
+	for _, step := range f.Steps {
+		for _, widget := range step.Widgets {
+
+			// Unpack the question properties.
+			props := widget.Props.(bson.M)
+
+			// We are looking to only include submissions with includeInGroups or groupSubmissions.
+			gs, gsok := props["groupSubmissions"]
```

Comment:
These variable names aren't the most readable to me.  maybe `incInGroups` or something different.  Also, it would be nice to just use a straight `ok` here for both checks.